### ISSUE TITLE
Use of '_' as an identifier might not be supported in releases after Java SE 8

### ISSUE THREAD

**Issue by leonard84** [spockframework/spock#69]
When compiling Spock with JDK 8 the compiler gives this warning message:
```
Specification.java:46: warning: '_' used as an identifier
  public static final Object _ = Wildcard.INSTANCE;
                             ^
  (use of '_' as an identifier might not be supported in releases after Java SE 8)
```
Will this become an issue for Spock in the future with Java 9+, and if so what should we do about it.

**Comment by jaredsburrows** [comment]
Out of curiosity, can you post the source of that class and your `build.gradle`?

**Comment by leonard84** [comment]
@jaredsburrows you can see everything in this repo the file is Specification.java

**Comment by jaredsburrows** [comment]
Alright, so you see this when you runt unit tests

I am currently using:

    dependencies {
        testCompile 'org.codehaus.groovy:groovy-all:2.4.3'
        testCompile 'org.spockframework:spock-core:1.0-groovy-2.4'
    }

Along with:

    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8

And everything is running and working just fine.

**Comment by leonard84** [comment]
Yes, it works fine with JDK 8 the warning states that it may change with JDK 9 or later

`(use of '_' as an identifier might not be supported in releases after Java SE 8)`

**Comment by jaredsburrows** [comment]
I just never get those warnings at all. How are you seeing those? I am using the latest `Spock` and `Groovy` as well as `JDK8`?

**Comment by leonard84** [comment]
See http://openjdk.java.net/jeps/213
> [...], using underscore ("_") as an identifier, which generates a warning as of Java SE 8, should be turned into an error in Java SE 9. It is also proposed that interfaces be allowed to have private methods.

**Comment by jaredsburrows** [comment]
I have never been able to reproduce this.

**Comment by leonard84** [comment]
@jaredsburrows it does not appear when you use spock, you need to build spock with jdk8.

**Comment by Abrasha** [comment]
@leonard84, @pniederw  I think this issue becomes more urgent with every day.

> Underscore ("_") is a keyword in Java 9. If you use "_" as an identifier, javac of JDK 8+ issues a warning. javac of JDK 9, issues an error! To check and migrate your Java code to avoid using "_" as an identifier

What could be a workaround for this?

**Comment by leonard84** [comment]
@Abrasha currently spock is using Java 6 source-level where it is still allowed, and even if we switch to Java 8 it will still be allowed. I'm fairly certain that you can run pre-Java 9 code with a Java 9 runtime. And if everything fails we will have to move it to a groovy class, since groovy source is currently Java 5 source-level equivalent.

**Comment by Abrasha** [comment]
@leonard84 thank you for explanation. It makes sense. I have not faced this issue yet. Just interesting how it could be solved in future

**Comment by rabestro** [comment]
Unfortunately, Java 9+ doesn't support `_` anymore as an identifier. 

<img width="574" alt="image" src="https://user-images.githubusercontent.com/52966251/206869361-613e8b8b-7885-464a-ab73-641e08b86ab2.png">
 
How about double underscore?

```java
public class Main {
    public static void main(String[] args) {
        var __ = "Hello, World!";
        System.out.println(__);
        
        var $ = "Hello, Spock!";
        System.out.println($);
    }
}
```

**Comment by Vampire** [comment]
Yeah, but still, what problem do you see @rabestro?
As Leonard already described above, Spock is compiled for Java 8 so it is still usable and even if not it could be written as Groovy class as long as Groovy still supports it.
That you cannot define such a variable in a project just using Spock when using a newer language version is irrelevant.
